node/network/collator-protocol/src/lib.rs

```diff
@@ -134,3 +166,53 @@ async fn modify_reputation(
 
 	sender.send_message(NetworkBridgeTxMessage::ReportPeer(peer, rep)).await;
 }
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum ProspectiveParachainsMode {
+	// v2 runtime API: no prospective parachains.
+	Disabled,
+	// vstaging runtime API: prospective parachains.
+	Enabled,
+}
+
+impl ProspectiveParachainsMode {
+	fn is_enabled(&self) -> bool {
+		matches!(self, Self::Enabled)
+	}
+}
+
+async fn prospective_parachains_mode<Sender>(
+	sender: &mut Sender,
+	leaf_hash: Hash,
+) -> Result<ProspectiveParachainsMode, error::Error>
+where
+	Sender: polkadot_node_subsystem::CollatorProtocolSenderTrait,
+{
+	// TODO: call a Runtime API once staging version is available
```

Comment:
Is this TODO actually necessary? The current implementation looks correct to me.

Reply:
At the moment of writing it we didn't agree on keeping things this way.
I was actually planning to submit a PR to extract this into utils along with MAX_DEPTH